vendor/parser-extended/parser.mly

```diff
@@ -85,15 +89,20 @@ let psig_typext (te, ext) =
   (Psig_typext te, ext)
 let psig_value (vd, ext) =
   (Psig_value vd, ext)
-let psig_type ((nr, ext), tys) =
-  (Psig_type (nr, tys), ext)
-let psig_typesubst ((nr, ext), tys) =
+*)
+
+let psig_type (nr, tys) =
+  Psig_type (nr, tys)
+let psig_typesubst (nr, tys) =
   assert (nr = Recursive); (* see [no_nonrec_flag] *)
-  (Psig_typesubst tys, ext)
+  Psig_typesubst tys
```

Comment:
The use of `psig_typesubst` looks like
```ocaml
| type_subst_declarations
        { psig_typesubst $1 }
```

This is not substitutable with `Psig_typesubst $1` because `$1` is a tuple.

We could comment them but this would result in a larger diff later in this file. This applies to all the "tuple-taking" functions here.

Reply:
The patch is indeed large and will make backports harder. I'll merge this PR knowing that:
- The new representation for extensions doesn't make sense upstream or in a CST library.
- The next backport will require some more work.